src/interfaces/SpokePool.ts

```diff
@@ -25,8 +25,12 @@ export interface Deposit {
 }
 
 export interface DepositWithBlock extends Deposit, SortableEvent {
-  blockTimestamp: number;
   quoteBlockNumber: number;
+  blockTimestamp?: number;
+}
+
+export interface UBADepositWithBlock extends DepositWithBlock {
+  blockTimestamp: number;
 }
```

Comment:
Rather than adding another type, is it worth just setting `blockTimestamp` in `DepositWithBlock` to 0 for pre-UBA scenarios? It's a bit clunky, but it might be nice to avoid all the conditional typecasting.

Reply:
yeah i originally wanted to avoid creating problems where a blockTimestamp is set to 0 but is understood to be something else by another client, but i think the overall changes we have to make to support that aren't worth it